core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -482,47 +486,65 @@ class CityStats {
-            newFinalStatList["Excess food to production"] =
-                    Stats().apply { production = totalFood; food = -totalFood }
+        if (totalFood > 0 && constructionMatchesFilter(currentConstruction, "Excess Food converted to Production when under construction")) {
+            newFinalStatList["Excess food to production"] = Stats().apply { production = totalFood; food = -totalFood }
         }
 
         if (cityInfo.isInResistance())
             newFinalStatList.clear()  // NOPE
 
         if (newFinalStatList.values.map { it.production }.sum() < 1)  // Minimum production for things to progress
-            newFinalStatList["Production"] = Stats().apply { production = 1F }
+            newFinalStatList["Production"] = Stats().apply { production = 1f }
         finalStatList = newFinalStatList
     }
 
+    private fun getCitySpecificUniques(): Sequence<Unique> {
+        return cityInfo.cityConstructions.builtBuildingUniqueMap.getAllUniques()
+        .filter { it.params.isNotEmpty() && it.params.last() == "in this city" }
+    }
+
+    private fun getUniquesForThisCity(
+        unique: String,
+        // We might have to cached to avoid concurrency problems, so if we don't, just get it directly
+        citySpecificUniques: Sequence<Unique> = getCitySpecificUniques()
+    ): Sequence<Unique> {
+        return citySpecificUniques.filter { it.placeholderText == unique } +
+                cityInfo.civInfo.getMatchingUniques(unique).filter { cityInfo.matchesFilter(it.params[1]) }
+    }
+
     private fun getBuildingMaintenanceCosts(citySpecificUniques: Sequence<Unique>): Float {
         // Same here - will have a different UI display.
         var buildingsMaintenance = cityInfo.cityConstructions.getMaintenanceCosts().toFloat() // this is AFTER the bonus calculation!
         if (!cityInfo.civInfo.isPlayerCivilization()) {
             buildingsMaintenance *= cityInfo.civInfo.gameInfo.getDifficulty().aiBuildingMaintenanceModifier
         }
 
-        // e.g. "-[50]% Maintenance costs [in this city]"
-        val maintenanceUniqueTemplate = "-[]% building maintenance costs []"
-        val maintenanceUniques = citySpecificUniques.filter { it.placeholderText == maintenanceUniqueTemplate } +
-                cityInfo.civInfo.getMatchingUniques(maintenanceUniqueTemplate).filter { cityInfo.matchesFilter(it.params[1]) }
-        for (unique in maintenanceUniques) buildingsMaintenance *= (1f - unique.params[0].toFloat() / 100)
+        // e.g. "-[50]% maintenance costs for buildings [in this city]"
+        for (unique in getUniquesForThisCity("-[]% maintenance cost for buildings []", citySpecificUniques)) {
+            buildingsMaintenance *= (1f - unique.params[0].toFloat() / 100)
+        }
+
+        // Deprecated since 3.15
+            for (unique in getUniquesForThisCity("-[]% building maintenance costs []", citySpecificUniques)) {
```

Comment:
I'm pretty sure you created this unique, no? Then you can just delete it

Reply:
According to git blame, this was here before